src/Adapter/MSTestAdapter.PlatformServices/Services/ExecutionContextService.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+
+namespace Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices;
+
+internal static class ExecutionContextService
+{
+    /// <summary>
+    /// The execution context to use by class. This context will be derived from the assembly level context and shared down
```

Comment:
When you say derived here, what do you mean? That it will by a copy of the thread context that was used to run Assembly setup?

Reply:
Yes sorry poor wording. Also the alg changed a bit, we are trying to retrieve the cotnext left by `AssemblyInitialize` method, if it doesn't exist (suppressed or not such method) we will capture current state (i.e. main execution thread).